### ISSUE TITLE
Grab all keyboard events: Should be off when Remmina loses focus

### ISSUE THREAD

**Issue by bernd-wechner** [FreeRDP/Remmina#995]
Context:
* Client (OS name and version): Linux Mint 18
* Remmina version (remmina --version): 1.2.0-rcgit-15
* Desktop environment (GNOME, Unity, KDE, ..):  Cinnamon
* Connecting to (OS and version): Windows 7 and Linux Mint 18 (matters not, remote OS irrelevant to issue)
* Connecting via (RDP, VNC, ...): RDP and VNC (matters not, protocol irrelevant to issue)

Issue:
When I enable Grab all keyboard events, this is great because I can ALT+TAB on the remote system. Love it! Alas if I move to another window (and/or workspace) and forget I have this set, bizarreness is encountered. Not only does keyboard no longer seem to work (in reality it is, just all keystrokes going to Remmina and I forgot ;-) but also menus don't drop down on mouse clicks and other bizarre issue (which may be MINT/Cinnamon specific).

Point though is, that when the Remmina window that has Grab all keyboard events set loses focus it should stop grabbing them ...

**Comment by antenore** [comment]
Hi Bernard.

I cannot reproduce it in my environments strangely... That said, yes, we should turn it off. We will put this in our list (I hope we will have time soon).

May I ask you to tests other DE than cinnamon please?

**Comment by lutzfischer** [comment]
I have a related problem
Remmina grabs the keyboard on mouse-hover. I have a dual screen setup and very often have one remote desktop open on one screen and some local work on another.
Regularly I find I type things in the remote desktop session even so the actual active window is a local editor. This is a mayor annoyance with an otherwise great software. 

Steps to reproduce:
have click to focus mode active 
have a remote desktop session open
open notepad there
make a terminal (or anything else) the active window
move the mouse over the remmina-remote desktop window
start typing

you will type into notepad on the remote machine - even so the terminal has the focus.

I have remmina 1.1.2 from the debian repository.

**Comment by giox069** [comment]
@lutzfischer: we cannot patch remmina 1.1.2. You should try Remmina 1.2.X, but we currently we don't have packages for Debian, so you should recompile remmina by hand to have remmina 1.2. In the meanwhile we are working to setup a debian packaging system, but I cannot give you more precise informations.

**Comment by bernd-wechner** [comment]
@antenore Alas not easy for me to try another DE. Can put it on the list but amazingly busy this end. I wanted to try and build Remmina some while ago but on Mint 17.2 lots of stuff was awry, as it was based on a dated Ubuntu stem and I'm hoping now on Mint 18 I'll be able to build it. Then it's not out of the question that I look into patching it myself if I can nail the issue. But time is very very scarce alas with numerous projects pulling my every limb assunder. 

Instead, what I can do which is only half as good but still indicative, is test Remmina so I just did. Switched over to it, and clicked the "Grab all keyboard events" button. Puzzling behaviour is demonstrated but interesting and possibly informative for you.

As I just did it, while typing this I can type out the steps for you easily. Context is, I am working on one Workspace and Remmina is open with a running session on my HTPC on another workspace. I switch work spaces with ALT+CTRL+ARROW (right and left) and this is what I did:

1. Switch to Remmina Workspace
2. Click the  "Grab all keyboard events" button.
3. Switch back this workspace (and to my web browser)
4. Type - it works. Keyboard events aren't being grabbed!

Awesome! So ...

5. Switch back to Remmina Workspace.
6. Press ALT+TAB to flick through windows. I'm cycling through local windows.
7. Click in the Remmina window (remote session)
8. Press ALT+TAB to flick through windows. I'm cycling through local windows. I'm no cycling through the remote session windows. 
9. Switch back to this workspace. Doesn't work, is swichting on remote system (both are Mint 18)
10. Switch back to this workspace by clicking on workspace icon in my tasbar
11. Try typing in this box. No ressponse, keyboard events seemingly being swallows by Remmina. 
12. Switch back to Remmina workspace. 
13. Click the  "Grab all keyboard events" button.
14. Switch back to here. I can type again. 

In summary, it seems only to start grabbing the keyboard events once the remote session has focus, and then holds onto it. 

Can maybe try more rigorous tests later, but it's way past midnight now, just saw a pile of guests off, and have to hit the sack as my 3 year old will be getting me up early ;-). Just a  wind down moment in which I got notification of your reply.

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @bernd-wechner @giox069 @lutzfischer
